### PR TITLE
Turn `.container` into a utility component

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
Devs/maintenance

## What is it doing for them?
- One step closer to getting rid of utility classes, see #10018 for more details
- I had to add `isContainer` to `Space` as `Container` and `Space` are both styled components and I couldn't find another way to merge them - if we hate it, let's discuss.
- Something is up with styled-components and type declaration where it ruins the rendering of the rest of the file (see screenshots). If we move that type declaration in its own right, it renders it properly. Extra faff, but so much more readable.
Screenshot 2023-07-06 at 11 34 00
Screenshot 2023-07-06 at 11 34 10

### CODE REVIEW

File: catalogue/webapp/components/WorkDetailsText/WorkDetailsText.test.tsx [hunk 1]
```diff
@@ -28,7 +28,11 @@ describe('WorkDetailsText', () => {
       <WorkDetailsText
         contents={
           <>
-            This is a <ul>React</ul> element
+            This is a{' '}
+            <ul>
+              <li>React</li>
```
Comment (rcantin-w) [inline catalogue/webapp/components/WorkDetailsText/WorkDetailsText.test.tsx:33]: This was just a linting complaint

File: content/webapp/pages/stories/index.tsx [hunk 2]
```diff
@@ -51,11 +52,32 @@ const ArticlesContainer = styled.div`
   background-color: ${props => props.theme.color('warmNeutral.300')};
 `;
 
-const StoryPromoContainer = styled.div.attrs({
-  className: 'container container--scroll',
-})`
+const StoryPromoContainer = styled(Container)`
   -webkit-overflow-scrolling: touch;
 
+  /* former .container--scroll */
```
Comment (rcantin-w) [inline content/webapp/pages/stories/index.tsx:58]: AFAIK this is the only place it was used. I think @AMBacelar is working on adding it in Search as well, but I still feel it doesn't need to be a utility class. Could maybe be it's own component, or something else.

**Review by gestchild — APPROVED** [review]
I don't love the isContainer, but I couldn't find a better solution either

**Comment by rcantin-w** [comment]
> I don't love the isContainer

Same, I really don't like it!
I just realised there might be [a simpler solution](https://github.com/wellcomecollection/wellcomecollection.org/pull/10019/commits/b0a75a8657baca2fbcf65f46dc90c6afce30d26d?diff=unified&w=1), it looks the same to me as prod does, so idk if I was just too close to it when I originally made the PR? Do confirm if I'm missing something?

**Comment by gestchild** [comment]
Ha, that would work. It did actually cross my mind when I first looked at it and then I got caught up in trying to find a way to combine styled components and forgot.

**Comment by rcantin-w** [comment]
Think that's what happened to me too 😅 sometimes we just have to walk away and come back. I'll merge on Monday 👍

**Pull request merged**